source/forth3/src/vm/mod.rs

```diff
@@ -44,16 +47,18 @@ pub struct Forth<T: 'static> {
     async_builtins: &'static [AsyncBuiltinEntry<T>],
 }
 
-enum ProcessAction {
-    Continue,
-    Execute,
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+
+pub enum InterpretAction {
```

Comment:
I'm not particularly attached to the name `InterpretAction`, but I couldn't think of something better --- @jamesmunns, if you want to bikeshed this, I'm all ears!

Reply:
I don't have a very strong opinion on the name, though it would probably be good to add doc comments here that explain what each choice means.

For `AcceptInput` (which I think means "block until input buffer is filled"), do we want to make this something like:

```rust
pub enum InterpretAction {
    Done,
    Continue,
    BlockedOn(BlockReason),
}

pub enum BlockReason {
    AcceptInput,
    OutputFull,
    // ...
    Custom(u32), // User defined blocking reason?
}
```

Reply:
whoops, I meant to add docs to this --- thanks for calling that out!

Reply:
we could add a nested enum of blocking reasons, but the one you're proposing does make the return value substantially larger: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2a13f71d45fd5a02872bd972c1ecba58

worrying about enum size here may be a bit of a premature optimization, but it *is* the return value of basically every interpreter step, so we pass these enums around kind of a lot --- and they're already inside of a `Result<InterpretAction, Error>`, which adds a couple more layers of discriminants...we probably want to ensure that the return value of these functions fits in a register, ideally?